src/legacy/core_plugins/kibana/public/dashboard/np_ready/url_helper.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { parse } from 'url';
+import { absoluteToParsedUrl } from '../legacy_imports';
+import { DashboardConstants } from './dashboard_constants';
+/**
+ * Return query params from URL
+ * @param url given url
+ */
+export function getUrlVars(url: string): Record<string, string> {
+  const vars: Record<string, string> = {};
+  // @ts-ignore
+  url.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(_, key, value) {
```

Comment:
That's an interesting usage of `replace`. However I think we can use `matchAll` in this scenario:
```ts
for ([_, key, value] of url.matchAll(/[?&]+([^=&]+)=([^&]*)/gi)) {
 vars[key] = value;
}
```

Then you don;t need any `ts-ignore`s.

Reply:
This was taken from here:
https://github.com/elastic/kibana/blob/8e9a8a84dccfa7965ce8a22362885e6cdef8b51f/x-pack/legacy/plugins/ml/public/application/jobs/jobs_list/components/utils.js#L374

So  I assumed it was well-tested. But I like any suggestion that gets rid of ts-ignore directives 👍